lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2043,68 +2043,66 @@ quitStakePool ctx (ApiT wid) body = do
 
     genChange = delegationAddress @n
 
--- More testable helper for `listStakeKeys`.
+-- | More testable helper for `listStakeKeys`.
 listStakeKeys'
-    :: forall (n :: NetworkDiscriminant) m. Monad m
-    => UTxO.UTxO
-        -- ^ The wallet's UTxO
-    -> (Address -> Maybe RewardAccount)
+    :: forall (n :: NetworkDiscriminant)
+     . (Address -> Maybe RewardAccount)
         -- ^ Lookup reward account of addr
-    -> (Set RewardAccount -> m (Map RewardAccount (Maybe Coin)))
-        -- ^ Batch fetch of rewards
-    -> [(RewardAccount, Natural, ApiWalletDelegation)]
-        -- ^ The wallet's known stake keys, along with derivation index, and
-        -- delegation status.
+    -> UTxO.UTxO
+        -- ^ The wallet's UTxO
+    -> [(RewardAccount, Natural)]
+        -- ^ The wallet's known stake keys, along with derivation index
         --
         -- The `RewardAccount`s (and the derivation indices) must be unique.
-    -> m (ApiStakeKeys n)
-listStakeKeys' utxo lookupStakeRef fetchRewards ourKeysWithInfo = do
-        let distr = stakeKeyCoinDistr lookupStakeRef utxo
-        let stakeKeysInUTxO = catMaybes $ Map.keys distr
-        let stake acc = fromMaybe (Coin 0) $ Map.lookup acc distr
-
-        let ourKeys = map (\(acc,_,_) -> acc) ourKeysWithInfo
-
-        let allKeys = ourKeys <> stakeKeysInUTxO
-
-        -- If we wanted to know whether a stake key is registered or not, we
-        -- could look at the difference between @Nothing@ and
-        -- @Just (Coin 0)@ from the response here, instead of hiding the
-        -- difference.
-        rewardsMap <- fetchRewards $ Set.fromList allKeys
-
-        let rewards acc = fromMaybe (Coin 0) $
-                join $ Map.lookup acc rewardsMap
-
-        let mkOurs (acc, ix, deleg) = ApiOurStakeKey
-                { _index = ix
-                , _key = (ApiT acc, Proxy)
-                , _rewardBalance = coinToQuantity $
-                    rewards acc
-                , _delegation = deleg
-                , _stake = coinToQuantity $
-                    stake (Just acc) <> rewards acc
-                }
+    -> [(RewardAccount, ApiWalletDelegation)]
+        -- ^ Delegation status
+    -> [(RewardAccount, Coin)]
```

Comment:
Is taking 
```
-> [(RewardAccount, Natural)]
-> [(RewardAccount, ApiWalletDelegation)]
-> [(RewardAccount, Coin)]
```
which all have to match really simpler?

Reply:
This was just a draft that I sketched and accidentally pushed to the wrong branch.
It would be better to have a record type or n-tuple.